File: app/src/main/java/com/example/schoolairdroprefactoredition/scene/ssb/fragment/SellingViewModel.java
package com.example.schoolairdroprefactoredition.scene.ssb.fragment;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.schoolairdroprefactoredition.domain.DomainGoodsInfo;
import com.example.schoolairdroprefactoredition.scene.main.base.BaseStateViewModel;
import com.example.schoolairdroprefactoredition.presenter.callback.ISellingCallback;
import com.example.schoolairdroprefactoredition.presenter.impl.SellingImpl;

public class SellingViewModel extends BaseStateViewModel implements ISellingCallback {

    private SellingImpl sellingImpl;

    private MutableLiveData<DomainGoodsInfo> mSellingBeans = new MutableLiveData<>();

    public SellingViewModel() {
        sellingImpl = new SellingImpl();
        sellingImpl.registerCallback(this);
    }

    public LiveData<DomainGoodsInfo> getSelling(String token) {
        sellingImpl.getSelling(token);
        return mSellingBeans;
    }

    @Override
    public void onSellingLoaded(DomainGoodsInfo beans) {
        mSellingBeans.postValue(beans);
    }

    @Override
    public void onError() {
        mOnRequestListener.onError();
    }

}

File: app/src/main/java/com/example/schoolairdroprefactoredition/viewmodel/SellingViewModel.kt
package com.example.schoolairdroprefactoredition.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.schoolairdroprefactoredition.domain.DomainSelling
import com.example.schoolairdroprefactoredition.repository.SellingRepository

class SellingViewModel : ViewModel() {

    private val sellingPage = 0

    private val postsPage = 0

    private val sellingRepository = SellingRepository.getInstance()

    /**
     * 下架物品
     */
    fun deleteGoods(token: String, goodsID: String): LiveData<Boolean> {
        val deleteGoodsResultLiveData = MutableLiveData<Boolean>()
        sellingRepository.deleteGoods(token, goodsID) {
            deleteGoodsResultLiveData.postValue(it)
        }
        return deleteGoodsResultLiveData
    }

    /**
     * 获取用户在售物品
     */
    fun getSelling(userID: Int): LiveData<DomainSelling?> {
        val sellingLiveData = MutableLiveData<DomainSelling?>()
        sellingRepository.getSelling(userID) {
            sellingLiveData.postValue(it)
        }
        return sellingLiveData
    }
}